src/main/java/org/openlmis/upload/TradeItemService.java

```diff
@@ -150,7 +150,31 @@ public boolean updateResource(JsonObject jsonObject, String id) {
     JsonObjectBuilder builder = Json.createObjectBuilder();
     jsonObject.forEach(builder::add);
     builder.add(ID, tradeItemId);
-    return super.updateResource(builder.build(), "");
+    
+    String url = buildUpdateUrl(configuration.getHost() + getUrl(), "");
```

Comment:
Instead of copy/pasting this whole method from BaseCommunicarionService, can we update the update resource method to conditionally override id? We can keep the current method to override it by default, and introduce an overloaded method with an additional flag - addIdToObject

Reply:
Agreed. I have changed that method.